3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -279,14 +282,20 @@ object EnrichmentManager {
       case f => f
     }
 
-    // Fetch IAB enrichment context (before anonymizing the IP address)
+    // Fetch IAB enrichment context (before anonymizing the IP address).
+    // IAB enrichment is called only if the IP is v4 (and after removing the port if any)
+    // and if the user agent is defined.
     val iabContext = registry.getIabEnrichment match {
       case Some(iab) =>
-        iab
-          .getIabContext(Option(event.useragent),
-                         Option(event.user_ipaddress),
-                         Option(event.derived_tstamp).map(EventEnrichments.fromTimestamp))
-          .map(_.some)
+        event.user_ipaddress match {
+          case IPv4Regex(ipv4) if !List(null, "", s"\0").contains(event.useragent) =>
```

Comment:
```diff
-- s"\0"
++ "\0"
```

also are those checks necessary given the regex match?

also, I'd rather we do `Option(event.user_ipaddress).map ...`

Reply:
I'll update `"\0"`.

The regex is for the IP, while the checks are for the user agent.

Reply:
I can't remove the `s` :
```
Octal escape literals are unsupported, use \u0000 instead
```